### PR TITLE
Omar add routing check

### PR DESCRIPTION
**Pull request by omarcodex**
For next week: goal is to highlight current tab upon direct URL navigation.

### CODE REVIEW

File: scripts/login.js [hunk 1]
```diff
@@ -163,9 +163,15 @@ function retrieveUrl(location) {
   }
 }
 
-function setLandingTab(URL) {
+function setLandingTab(input) {
```
Comment (JohannBillar) [inline scripts/login.js]: The URL parameter name was more descriptive to me, what was your reason for changing?

File: scripts/login.js [hunk 2]
```diff
@@ -154,26 +154,26 @@ function checkDefaultDoWhop(person) {
 
 // For checking the pre-specified routing location. New.
 function retrieveUrl(location) {
-  if (location.match(/#(.+)/)[1]) {
+  if (location.match(/#(.+)/) != null && location.match(/#(.+)/)[1] != null) {
```
Comment (JohannBillar) [inline scripts/login.js]: Be mindful of the implicit coercion that's happening with == equality.

File: scripts/login.js [hunk 3]
```diff
@@ -154,26 +154,26 @@ function checkDefaultDoWhop(person) {
...
     var y = location.match(/#(.+)/)[1];
-    console.log('current location is', y);
     return y;
   } else {
     return null;
```
Comment (JohannBillar) [inline scripts/login.js]: 🐞 When I sign in for the first time. retrieveUrl returns null. Then, setLandingTab is invoked with input set to null resulting in a TypeErrror: Cannot read property 'match' of null.

**Comment by JohannBillar** [comment]
Nice work @omarcodex! I like the way you build this. 💯  
I ran into 2 issues, maybe edge cases because I'm not using the feature as designed:
- I found a TypeError (see code review comments). 
- When I sign in for the first time, or after a hard refresh of the client and then sign in, the currentTab stored in the session for my UID doesn't appear to be read from the database because the tab is not selected in my UI. Then when I select a tab, sign out, sign in, now the currentTab is read from the database. Hope that made sense.  😃

File: scripts/login.js (hunk 1 above)
Comment (omarcodex) [inline scripts/login.js]: It was changed to be more generic for anchor point or URL input - solution now is to say `href` :)

File: scripts/login.js [hunk 4]
```diff
@@ -48,10 +48,9 @@
     retrieveMyDoWhops(user.uid);
 
     // We are checking if there is a direct routing to the site.
-    // TO-DO: Alternative option is to restore a "saved session."
     if (window.location.href.match(/#(.+)/)) {
       setLandingTab(retrieveUrl(window.location.href)); // Check.
-    }
+    } // TO-DO: Alternative option is to restore a "saved session."
```
Comment (JohannBillar) [inline scripts/login.js]: 👍

**Pull request merged**